File: functions/models/locations/supported-town-or-institution.js
/* eslint-disable*/
class SupportedTownOrInstitution {

    constructor(cityFK, townOrInstitutionName, townOrInstitutionNo) {
        this.cityFK = cityFK;
        this.townOrInstitutionName = townOrInstitutionName;
        this.townOrInstitutionNo = townOrInstitutionNo;
    }

    toJson() {
        return {
            cityFK: this.cityFK,
            townOrInstitutionName: this.townOrInstitutionName,
            townOrInstitutionNo: this.townOrInstitutionNo,
        };
    }
}

export default SupportedTownOrInstitution 

File: lib/models/locations/supported_town_or_institution.dart
// Collection Name /supported_towns_or_institutions/{townOrInstitutionId}

import '../converter.dart';
import 'town_or_institution.dart';

class SupportedTownOrInstitution {
  String townOrInstitutionNo;
  String cityFK;
  TownOrInstitution townOrInstitutionName;

  SupportedTownOrInstitution({
    this.townOrInstitutionNo = '5',
    this.cityFK = '1',
    this.townOrInstitutionName = TownOrInstitution.mayville,
  });

  Map<String, dynamic> toJson() => {
        "townOrInstitutionNo": townOrInstitutionNo,
        "cityFK": cityFK,
        "townOrInstitutionName":
            Converter.townOrInstitutionAsString(townOrInstitutionName),
      };

  factory SupportedTownOrInstitution.fromJson(dynamic json) =>
      SupportedTownOrInstitution(
          townOrInstitutionNo: json['townOrInstitutionNo'],
          cityFK: json['cityFK'],
          townOrInstitutionName:
              Converter.toTownOrInstitution(json['townOrInstitutionName']));

  @override
  String toString() {
    return Converter.townOrInstitutionAsString(townOrInstitutionName);
  }
}
